accounting/rewind.go

```diff
@@ -71,15 +72,18 @@ func AccountAtRound(account models.Account, round uint64, db idb.IndexerDb) (acc
 			if addr == stxn.Txn.Receiver {
 				acct.AmountWithoutPendingRewards -= uint64(stxn.Txn.Amount)
 				acct.AmountWithoutPendingRewards -= uint64(stxn.ReceiverRewards)
+				acct.Rewards -= uint64(stxn.ReceiverRewards)
 			}
 			if addr == stxn.Txn.CloseRemainderTo {
 				// unwind receiving a close-to
 				acct.AmountWithoutPendingRewards -= uint64(stxn.ClosingAmount)
 				acct.AmountWithoutPendingRewards -= uint64(stxn.CloseRewards)
+				acct.Rewards -= uint64(stxn.CloseRewards)
 			} else if !stxn.Txn.CloseRemainderTo.IsZero() {
 				// unwind sending a close-to
 				acct.AmountWithoutPendingRewards += uint64(stxn.ClosingAmount)
-				acct.AmountWithoutPendingRewards += uint64(stxn.CloseRewards)
+				// TODO: this doesn't seem right. Rewards don't come from the account balance so what is this doing?
```

Comment:
No, wait, CloseRewards ... go to the close receiver? Now I'm confused and will need to re-read the impl in algod; but now I'm guessing this is pending rewards on the account that weren't paid out before it was closed. The virtual sequence would be: apply pending rewards, pay amount receiver, everything else to close-to addr.

Reply:
I haven't verified this. If there are "CloseRewards" then the CloseToAddress must have been set and the Sender account needs to be empty at the end of the transaction. Therefore "SenderRewards" had better be equal to zero, I'm having a hard time coming up with a scenario where CloseRewards and SenderRewards are both non zero.

I'm pretty convinced that this was a bug and removing the line is correct. Rewards always originate from the RewardsPool. Therefore when rewinding transactions rewards should never be added to an account other than the RewardsPool. Since we shouldn't be rewinding the RewardPool account we should never be adding CloseRewards/ReceiverRewards/SenderRewards to an account in this function.

Reply:
Looking at algod payment.go and eval.go, CloseRewards is what is accrued on the CloseRemainderTo account which gets added in at this time just because the account is being touched. It shouldn't go to the Sender, that was a bug. ClosingAmount is everything left over after normal payment Amount and Fee.
Now the subtle bug might be that I think `ClosingAmount = previousBalance + SenderRewards - Fee - Amount` and we might be double-rewinding SenderRewards by doing it separately _and_ as ClosingAmount.

Reply:
Looking again again I think the Sender previous balance is okay with a closeremainderto. algebra on the previous comment:
`sender.previousBalance = +ClosingAmount + Fee + Amount - SenderRewards`
and I see all those pieces here.